src/plugins/intel_cpu/src/utils/jit_kernel.hpp

```diff
@@ -577,51 +837,82 @@ void jit_kernel::foreach(const B & begin,
     add(idx, step);
     jmp(loop, T_NEAR);
     L(exit);
-
-    free<Reg64>(idx);
 }
 
 template<typename T>
 jit_kernel::variable<T> jit_kernel::var() {
     using reg_type = typename reg_traits<T>::type;
     const auto & reg = reserve<reg_type>();
-    return variable<T>(*this, reg);
+    return variable<T>(*this, internal::make_shared(reg, *this));
+}
+
+template<typename T>
+jit_kernel::variable<T> jit_kernel::var(const T & val) {
+    using reg_type = typename reg_traits<T>::type;
+    const auto & reg = reserve<reg_type>();
+    variable<T> res(*this, internal::make_shared(reg, *this));
+    res = val;
+    return std::move(res);
+}
+
+template<typename T>
+const T & jit_kernel::constant(const T & c) {
+    auto res = _consts.store(&c, sizeof c);
+    return *reinterpret_cast<const T*>(res);
+}
+
+template<typename T>
+const T * jit_kernel::constant(const T * c, size_t size) {
+    auto res = _consts.store(c, size * sizeof(T));
+    return reinterpret_cast<const T*>(res);
 }
 
 template<typename T>
-jit_kernel::if_expression<T> jit_kernel::_if(const boolean_expression<T> & expr) {
+jit_kernel::if_expression<T> jit_kernel::_if(const boolean_expression<T> & expr) const {
     return if_expression<T>(expr);
 }
 
 namespace internal {
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// shared_reg
+
+template<typename Reg>
+shared_reg<Reg> make_shared(Reg & reg, jit_kernel & kernel) {
+    std::shared_ptr<Reg> ptr(&reg, [&kernel](Reg *preg) {
+        try {
+            kernel.free(*preg);
+        } catch(...) {}
```

Comment:
Why we do nothing in catch section?

Reply:
We can't throw exception form the destructor. I can only suppress exceptions. ):